src/reactor.h

```diff
@@ -0,0 +1,242 @@
+/// dynamically loaded by the Agent class when requested.
+///
+/// @section intro Introduction
+/// Place an introduction to the agent here.
+///
+/// @section agentparams Agent Parameters
+/// Place a description of the required input parameters which define the
+/// agent implementation.
+///
+/// @section optionalparams Optional Parameters
+/// Place a description of the optional input parameters to define the
+/// agent implementation.
+///
+/// @section detailed Detailed Behavior
+/// Place a description of the detailed behavior of the agent. Consider
+/// describing the behavior at the tick and tock as well as the behavior
+/// upon sending and receiving materials and messages.
+class Reactor : public cyclus::Facility  {
+ public:
+  /// Constructor for Reactor Class
+  /// @param ctx the cyclus context for access to simulation-wide parameters
+  explicit Reactor(cyclus::Context* ctx);
+
+  /// The Prime Directive
+  /// Generates code that handles all input file reading and restart operations
+  /// (e.g., reading from the database, instantiating a new object, etc.).
+  /// @warning The Prime Directive must have a space before it! (A fix will be
+  /// in 2.0 ^TM)
+
+  #pragma cyclus
+
+  #pragma cyclus note {"doc": "A stub facility is provided as a skeleton " \
+                              "for the design of new facility agents."}
+
+  /// A verbose printer for the Reactor
+  virtual std::string str();
+
+  /// Set up policies and buffers:
+  virtual void EnterNotify();
+
+  /// The handleTick function specific to the Reactor.
+  /// @param time the time of the tick
+  virtual void Tick();
+
+  /// The handleTick function specific to the Reactor.
+  /// @param time the time of the tock
+  virtual void Tock();
+
+  #pragma cyclus var { \
+    "doc": "Nameplate fusion power of the reactor", \
+    "tooltip": "Nameplate fusion power", \
+    "units": "MW", \
+    "uilabel": "Fusion Power" \
+  }
+  double fusion_power;
+
+  #pragma cyclus var { \
+    "doc": "Achievable system tritium breeding ratio before decay", \
+    "tooltip": "Achievable system tritium breeding ratio before decay", \
+    "units": "non-dimensional", \
+    "uilabel": "Tritium Breeding Ratio" \
+  }
+  double TBR;
+
+  #pragma cyclus var { \
+    "doc": "Minimum tritium inventory to hold in reserve in case of tritium recovery system failure", \
+    "tooltip": "Minimum tritium inventory to hold in reserve (excluding core invneotry)", \
+    "units": "kg", \
+    "uilabel": "Reserve Inventory" \
+  }
+  double reserve_inventory;
+
+  #pragma cyclus var { \
+    "doc": "Tritium core inventory required to start reactor", \
+    "tooltip": "Tritium core inventory required to start reactor", \
+    "units": "kg", \
+    "uilabel": "Start-up Inventory" \
+  }
+  double startup_inventory;
+
+  #pragma cyclus var { \
+    "default": 'fill', \
+    "doc": "Method of refueling the reactor", \
+    "tooltip": "Options: 'schedule' or 'fill'", \
```

Comment:
```suggestion
    "tooltip": "Options: 'schedule' or 'fill'", \
    "uitype": "combobox", \
    "categorical": ['schedule', 'fill'], \
```

Reply:
Did this one manually because I wasn't sure if it would break one of my unit tests, but this was implemented all the same. Thanks for spelling this one out, too, I couldn't quite figure out how to do it from just reading the documentation.